File: XcodeTemplates/Templates/Project Templates/iOS/Application Extension/Virtual Conference Provider Extension.xctemplate/VirtualConferenceProvider.m
//___FILEHEADER___

#import "___FILEBASENAME___.h"

@implementation ___FILEBASENAME___

- (void)fetchAvailableRoomTypesWithCompletionHandler:(void(^)(NSArray<EKVirtualConferenceRoomTypeDescriptor *> * _Nullable, NSError * _Nullable))completionHandler
{
    EKVirtualConferenceRoomTypeDescriptor *roomTypeDescriptor = [[EKVirtualConferenceRoomTypeDescriptor alloc] initWithTitle:@"My Virtual Conference Room" identifier:@"my_virtual_conference_room_identifier"];
    completionHandler(@[roomTypeDescriptor], nil);
}

- (void)fetchVirtualConferenceForIdentifier:(EKVirtualConferenceRoomTypeIdentifier)identifier completionHandler:(void(^)(EKVirtualConferenceDescriptor * _Nullable, NSError * _Nullable))completionHandler
{
    EKVirtualConferenceURLDescriptor *urlDescriptor = [[EKVirtualConferenceURLDescriptor alloc] initWithTitle:nil URL:[NSURL URLWithString:@"https://example.com/"]];
    EKVirtualConferenceDescriptor *virtualConferenceDescriptor = [[EKVirtualConferenceDescriptor alloc] initWithTitle:@"My Virtual Conference" URLDescriptors:@[urlDescriptor] conferenceDetails:@"Any other notes or details you wish to include about this virtual conference"];
    completionHandler(virtualConferenceDescriptor, nil);
}

@end

File: XcodeTemplates/Templates/Project Templates/iOS/Application Extension/Virtual Conference Provider Extension.xctemplate/VirtualConferenceProvider.swift
//___FILEHEADER___

import EventKit

class ___FILEBASENAME___: EKVirtualConferenceProvider {

    override func fetchAvailableRoomTypes(completionHandler: @escaping ([EKVirtualConferenceRoomTypeDescriptor]?, Error?) -> Void) {
        let roomType = EKVirtualConferenceRoomTypeDescriptor(title: "My Virtual Conference Room", identifier: "my_virtual_conference_room")
        completionHandler([roomType], nil)
    }
    
    override func fetchVirtualConference(identifier: EKVirtualConferenceRoomTypeIdentifier, completionHandler: @escaping (EKVirtualConferenceDescriptor?, Error?) -> Void) {
        let urlDescriptor = EKVirtualConferenceURLDescriptor(title: nil, url: URL(string: "https://example.com/")!)
        let virtualConferenceDescriptor = EKVirtualConferenceDescriptor(title: "My Virtual Conference", urlDescriptors: [urlDescriptor], conferenceDetails: "Any other notes or details you wish to include about this virtual conference.")
        completionHandler(virtualConferenceDescriptor, nil)
    }
}
